### ISSUE TITLE
Testing clearml-agent in K8s with GPU

### ISSUE THREAD

**Issue by jjamgmv** [allegroai/clearml-helm-charts#335]
### Describe the bug a clear and concise description of what the bug is.

When I execute this, I always receive this error. If I use the pip agent on the same machine, it works fine:
[notice] To update, run: python3.10 -m pip install --upgrade pip
/root/__start_agent__.sh: line 14:   684 Segmentation fault      (core dumped) $LOCAL_PYTHON -m clearml_agent execute --full-monitoring --require-queue --id 5c06200ada0f4aa2bfe9a16e0ebee84b

### What's your helm version?

v3.16.3

### What's your kubectl version?

v1.27.16 +rke2r2

### What's the chart version?

5.2.2

### Enter the changed values of values.yaml?

# -- Global parameters section
global:
  # -- Images registry
  imageRegistry: "docker.io"

# -- Private image registry configuration
imageCredentials:
  # -- Use private authentication mode
  enabled: false
  # -- If this is set, chart will not generate a secret but will use what is defined here
  existingSecret: ""
  # -- Registry name
  registry: docker.io
  # -- Registry username
  username: someone
  # -- Registry password
  password: pwd
  # -- Email
  email: someone@host.com

# -- ClearMl generic configurations
clearml:
  # -- If this is set, chart will not generate a secret but will use what is defined here
  existingAgentk8sglueSecret: ""
  # -- Agent k8s Glue basic auth key
  agentk8sglueKey: "xxx"
  # -- Agent k8s Glue basic auth secret
  agentk8sglueSecret: "xxx"

  # -- If this is set, chart will not generate a secret but will use what is defined here
  existingClearmlConfigSecret: ""
  # The secret should be defined as the following example
  #
  # apiVersion: v1
  # kind: Secret
  # metadata:
  #   name: secret-name
  # stringData:
  #   clearml.conf: |-
  #     sdk {
  #     }
  # -- ClearML configuration file
  clearmlConfig: |-
    sdk {
    }

# -- This agent will spawn queued experiments in new pods, a good use case is to combine this with
# GPU autoscaling nodes.
# https://github.com/allegroai/clearml-agent/tree/master/docker/k8s-glue
agentk8sglue:
  # -- Glue Agent image configuration
  image:
    registry: ""
    repository: "allegroai/clearml-agent-k8s-base"
    tag: "1.24-21"
  # -- Glue Agent number of pods
  replicaCount: 1

  # -- Glue Agent pod resources
  resources: {}

  # -- Glue Agent pod initContainers configs
  initContainers:
    # -- Glue Agent initcontainers pod resources
    resources: {}

  # -- if set, don't create a serviceAccountName but use defined existing one
  serviceExistingAccountName: ""

  # -- Check certificates validity for evefry UrlReference below.
  clearmlcheckCertificate: true

  # -- Reference to Api server url
  apiServerUrlReference: "https://api-clearml.xxx.com"
  # -- Reference to File server url
  fileServerUrlReference: "https://files-clearml.xxx.com"
  # -- Reference to Web server url
  webServerUrlReference: "https://app-clearml.xxx.com"

  # -- default container image for ClearML Task pod
  defaultContainerImage: clearml/fractional-gpu:u22-cu11.7-12gb
  # -- ClearML queue this agent will consume. Multiple queues can be specified with the following format: queue1,queue2,queue3
  queue: kollama
  # -- if ClearML queue does not exist, it will be create it if the value is set to true
  createQueueIfNotExists: false
  # -- labels setup for Agent pod (example in values.yaml comments)
  labels: {}
    # schedulerName: scheduler
  # -- annotations setup for Agent pod (example in values.yaml comments)
  annotations: {}
    # key1: value1
  # -- Extra Environment variables for Glue Agent
  extraEnvs: []
    # - name: PYTHONPATH
    #   value: "somepath"
  # -- container securityContext setup for Agent pod (example in values.yaml comments)
  podSecurityContext: {}
    #  runAsUser: 1001
    #  fsGroup: 1001
  # -- container securityContext setup for Agent pod (example in values.yaml comments)
  containerSecurityContext: {}
    #  runAsUser: 1001
    #  fsGroup: 1001
  # -- additional existing ClusterRoleBindings
  additionalClusterRoleBindings: []
    # - privileged
  # -- additional existing RoleBindings
  additionalRoleBindings: []
    # - privileged
  # -- nodeSelector setup for Agent pod (example in values.yaml comments)
  nodeSelector: {}
    # fleet: agent-nodes
  # -- tolerations setup for Agent pod (example in values.yaml comments)
  tolerations: []
  # -- affinity setup for Agent pod (example in values.yaml comments)
  affinity: {}
  # -- volumes definition for Glue Agent (example in values.yaml comments)
  volumes: []
    # - name: "yourvolume"
    #   nfs:
    #    server: 192.168.0.1
    #    path: /var/nfs/mount
  # -- volume mounts definition for Glue Agent (example in values.yaml comments)
  volumeMounts: []
    # - name: yourvolume
    #   mountPath: /yourpath
    #   subPath: userfolder

  # -- file definition for Glue Agent (example in values.yaml comments)
  fileMounts: []
    # - name: "integration.py"
    #   folderPath: "/mnt/python"
    #   fileContent: |-
    #     def get_template(*args, **kwargs):
    #       print("args: {}".format(args))
    #       print("kwargs: {}".format(kwargs))
    #       return {
    #           "template": {
    #           }
    #       }

  # -- base template for pods spawned to consume ClearML Task
  basePodTemplate:
    # -- labels setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    labels: {}
      # schedulerName: scheduler
    # -- annotations setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    annotations: {}
      # key1: value1
    # -- initContainers definition for pods spawned to consume ClearML Task (example in values.yaml comments)
    hostPID: true
    # initContainers:
    #   - name: train-container
    #     image: clearml/fractional-gpu:u22-cu11.7-12gb
    #     command: ['python3', '-c', 'print(f"Free GPU Memory: (free, global) {torch.cuda.mem_get_info()}")']
    # initContainers: []
      # - name: volume-dirs-init-cntr
      #   image: busybox:1.35
      #  command:
      #    - /bin/bash
      #    - -c
      #    - >
      #      /bin/echo "this is an init";
    # -- schedulerName setup for pods spawned to consume ClearML Task
    schedulerName: ""
    # -- volumes definition for pods spawned to consume ClearML Task (example in values.yaml comments)
    volumes: []
      # - name: "yourvolume"
      #   nfs:
      #    server: 192.168.0.1
      #    path: /var/nfs/mount
    # -- volume mounts definition for pods spawned to consume ClearML Task (example in values.yaml comments)
    volumeMounts: []
      # - name: yourvolume
      #   mountPath: /yourpath
      #   subPath: userfolder
    # -- file definition for pods spawned to consume ClearML Task (example in values.yaml comments)
    fileMounts: []
      # - name: "mounted-file.txt"
      #   folderPath: "/mnt/"
      #   fileContent: |-
      #     this is a test file
      #     with test content
    # -- environment variables for pods spawned to consume ClearML Task (example in values.yaml comments)
    env: []
      # # to setup access to private repo, setup secret with git credentials:
      # - name: CLEARML_AGENT_GIT_USER
      #   value: mygitusername
      # - name: CLEARML_AGENT_GIT_PASS
      #   valueFrom:
      #     secretKeyRef:
      #       name: git-password
      #       key: git-password
      # - name: CURL_CA_BUNDLE
      #   value: ""
      # - name: PYTHONWARNINGS
      #   value: "ignore:Unverified HTTPS request"
    # -- resources declaration for pods spawned to consume ClearML Task (example in values.yaml comments)
    resources:
      limits:
        nvidia.com/gpu: 1
    # -- priorityClassName setup for pods spawned to consume ClearML Task
    priorityClassName: ""
    # -- nodeSelector setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    nodeSelector:  
      nvidia.com/gpu.product: NVIDIA-L40S-SHARED
    #   fleet: gpu-nodes
    # -- tolerations setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    tolerations:
      - key: "nvidia.com/gpu"
        operator: Exists
        effect: "NoSchedule"
    # -- affinity setup for pods spawned to consume ClearML Task
    affinity: {}
    # -- securityContext setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    podSecurityContext: {}
      #  runAsUser: 1001
      #  fsGroup: 1001
    # -- securityContext setup for containers spawned to consume ClearML Task (example in values.yaml comments)
    containerSecurityContext: {}
      #  runAsUser: 1001
      #  fsGroup: 1001
    # -- hostAliases setup for pods spawned to consume ClearML Task (example in values.yaml comments)
    hostAliases: []
    # - ip: "127.0.0.1"
    #   hostnames:
    #   - "foo.local"
    #   - "bar.local"

# -- Sessions internal service configuration
sessions:
  # -- Enable/Disable sessions portmode WARNING: only one Agent deployment can have this set to true
  portModeEnabled: false
  # -- specific annotations for session services
  svcAnnotations: {}
  # -- service type ("NodePort" or "ClusterIP" or "LoadBalancer")
  svcType: "NodePort"
  # -- External IP sessions clients can connect to
  externalIP: 0.0.0.0
  # -- starting range of exposed NodePorts
  startingPort: 30000
  # -- maximum number of NodePorts exposed
  maxServices: 20

**Comment by jkhenning** [comment]
@jjamgmv can you provide the console log with the error?

**Comment by jjamgmv** [comment]
> [@jjamgmv](https://github.com/jjamgmv) can you provide the console log with the error?

+ export CLEARML_FILES_HOST=https://xxx-file-clearml.gmv.com
+ CLEARML_FILES_HOST=https://dsb-training-file-clearml.gmv.com
+ '[' -z https://xxx-file-clearml.gmv.com ']'
+ export CLEARML_FILES_HOST=https://xxx-file-clearml.gmv.com
+ CLEARML_FILES_HOST=https://xxx-file-clearml.gmv.com
+ export CLEARML_WEB_HOST=https://xxx-app-clearml.gmv.com
+ CLEARML_WEB_HOST=https://xxx-app-clearml.gmv.com
+ export CLEARML_API_HOST=https://dsb-training-api-clearml.gmv.com
+ CLEARML_API_HOST=https://xxx-api-clearml.gmv.com
+ echo https://dsb-training-file-clearml.gmv.com https://dsb-training-app-clearml.gmv.com https://dsb-training-api-clearml.gmv.com
https://dsb-training-file-clearml.gmv.com https://dsb-training-app-clearml.gmv.com https://dsb-training-api-clearml.gmv.com
+ '[' -z '' ']'
+ '[' -n '' ']'
+ python3 -m pip install -q -U clearml-agent
/usr/lib/python3/dist-packages/secretstorage/dhcrypto.py:15: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
  from cryptography.utils import int_from_bytes
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
+ QUEUE=dsb-training
+ MAX_PODS=2
+ EXTRA_ARGS='--namespace dsb-training-clearml-agent --template-yaml /root/template/template.yaml --ports-mode --num-of-services 2 --base-port 21000 --gateway-address 10.10.121.222  --create-queue '
+ echo 'api.credentials.access_key: xx'
/root/entrypoint.sh: line 28: /root/clearml.conf: Read-only file system
+ echo 'api.credentials.secret_key: xxxx--DhB92X9NiUn-A'
/root/entrypoint.sh: line 29: /root/clearml.conf: Read-only file system
+ echo 'api.api_server: https://dsb-training-api-clearml.gmv.com'
/root/entrypoint.sh: line 30: /root/clearml.conf: Read-only file system
+ echo 'api.web_server: https://dsb-training-app-clearml.gmv.com'
/root/entrypoint.sh: line 31: /root/clearml.conf: Read-only file system
+ echo 'api.files_server: https://dsb-training-file-clearml.gmv.com'
/root/entrypoint.sh: line 32: /root/clearml.conf: Read-only file system
+ ./provider_entrypoint.sh
+ source /root/.bashrc
++ '[' -z '' ']'
++ return
+ export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/bin:/root/bin
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/bin:/root/bin
+ [[ -z '' ]]
+ python3 k8s_glue_example.py --queue dsb-training --namespace dsb-training-clearml-agent --template-yaml /root/template/template.yaml --ports-mode --num-of-services 2 --base-port 21000 --gateway-address 10.10.121.222 --create-queue
/usr/local/lib/python3.6/dist-packages/clearml_agent/_vendor/jwt/utils.py:7: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
  from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurve
Traceback (most recent call last):
  File "k8s_glue_example.py", line 8, in <module>
    from clearml_agent.glue.k8s import K8sIntegration
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/glue/k8s.py", line 19, in <module>
    from clearml_agent.commands.events import Events
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/commands/__init__.py", line 3, in <module>
    from .worker import Worker
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/commands/worker.py", line 47, in <module>
    from clearml_agent.commands.base import resolve_names, ServiceCommandSection
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/commands/base.py", line 20, in <module>
    from clearml_agent.interface.base import ObjectID
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/interface/__init__.py", line 7, in <module>
    from .base import Parser, base_arguments, add_service, OnlyPluralChoicesHelpFormatter
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/interface/base.py", line 12, in <module>
    from clearml_agent.session import Session
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/session.py", line 23, in <module>
    from clearml_agent.helper.docker_args import DockerArgsSanitizer, sanitize_urls
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/helper/docker_args.py", line 18, in <module>
    from clearml_agent.helper.custom_template import CustomTemplate, TemplateResolver
  File "/usr/local/lib/python3.6/dist-packages/clearml_agent/helper/custom_template.py", line 46, in <module>
    class CustomTemplate(Template):
  File "/usr/lib/python3.6/string.py", line 74, in __init__
    cls.pattern = _re.compile(pattern, cls.flags | _re.VERBOSE)
  File "/usr/lib/python3.6/re.py", line 233, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.6/sre_parse.py", line 855, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File "/usr/lib/python3.6/sre_parse.py", line 765, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
  File "/usr/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File "/usr/lib/python3.6/sre_parse.py", line 765, in _parse
    p = _parse_sub(source, state, sub_verbose, nested + 1)
  File "/usr/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File "/usr/lib/python3.6/sre_parse.py", line 734, in _parse
    flags = _parse_flags(source, state, char)
  File "/usr/lib/python3.6/sre_parse.py", line 803, in _parse_flags
    raise source.error("bad inline flags: cannot turn on global flag", 1)
sre_constants.error: bad inline flags: cannot turn on global flag at position 92 (line 4, column 20)
stream closed EOF for dsb-training-clearml-agent/dsb-training-clearml-agent-78f5c55b55-ktl9j (k8s-glue)

**Comment by nielstenboom** [comment]
See https://github.com/clearml/clearml-agent/issues/247#issue-3263049939 for fix 

TLDR; set `CLEARML_AGENT_NO_UPDATE=true`
